File: inheritance.cpp
#include <iostream>
#include <string>
using namespace std;

class Car {
public:
    string brand;
    string model;
    int year;
    Car(string brand, string model, int year) {
        this->brand = brand;
        this->model = model;
        this->year = year;
    }
    void showDetails() const {
        cout << "Brand: " << brand << endl;
        cout << "Model: " << model << endl;
        cout << "Year: " << year << endl;
    }
};

class Toyota : public Car {
public:
    double price;
    
    Toyota(string model, int year, double price) : Car("Toyota", model, year) {
        this->price = price;
    }

    void showDetails() const {
        Car::showDetails();
        cout << "Price: " << price << endl;
    }
};

int main() {
    Toyota corolla("Corolla", 2010, 25000);
    corolla.showDetails();

    return 0;
}

File: inheritance.py
class Car:
    def __init__(self, brand, model, year):
        self.brand = brand
        self.model = model
        self.year = year

    def show_details(self):
        print(f'Brand: {self.brand}\nModel: {self.model}\nYear: {self.year}')




class Toyota(Car):
  def __init__(self, model, year, price):
    self.price = price
    super().__init__('Toyota', model, year)


car1 = Toyota('Corola', 2010, 25000)
car1.show_details()